std/parallelism.d

```diff
@@ -2864,10 +2864,10 @@ public:
             }
         }
 
-        ref T opIndex(size_t index)
+        ref inout(T) opIndex(size_t index) inout
```

Comment:
`inout` is difficult to understand so instead I'd stick with the newer `this` template parameter. The code would look like this:

```d
ref auto opIndex(this Qualified)(size_t index)
{
    assert(index < size, text(index, '\t', uint.max));
    return *(cast(typeof(front())*) (data + elemSize * index));
}
```